src/expander.c

```diff
@@ -0,0 +1,140 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   expander.c                                         :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: gwolf <gwolf@student.42vienna.com>         +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/07/07 13:07:02 by gwolf             #+#    #+#             */
+/*   Updated: 2023/07/14 19:37:36 by gwolf            ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+/**
+ * @file expander.c
+ * @brief Contains driver function to handle expansion and quote removal.
+ */
+#include "expander.h"
+
+/**
+ * @brief Search for token in input string and replace with replace.
+ *
+ * @param input String.
+ * @param pos Current position.
+ * @param token Struct containing the searched for token.
+ * @param replace Struct containing the replacement.
+ * @return t_error SUCCESS, ERR_MALLOC
+ */
+t_error	ft_insert_replace(char **input, size_t pos, t_tok token, t_tok replace)
+{
+	size_t	new_len;
+	char	*new_str;
+	char	*old_str;
+
+	new_len = ft_strlen(*input) - token.len + replace.len;
+	new_str = malloc(new_len);
+	if (!new_str)
+		return (ERR_MALLOC);
+	*new_str = '\0';
+	old_str = *input;
+	ft_strncat(new_str, old_str, pos);
+	ft_strcat(new_str, replace.str);
+	pos += token.len;
+	ft_strcat(new_str, (old_str + pos));
+	free(old_str);
+	*input = new_str;
+	return (SUCCESS);
+}
+
+/**
+ * @brief Remove single char from string. Doesn't realloc.
+ *
+ * @param input String.
+ * @param pos Current position.
+ * @return t_error SUCCESS
+ */
+t_error	ft_eat_char(char *input, size_t pos)
+{
+	char	*str1;
+	char	*str2;
+
+	str1 = input + pos;
+	str2 = str1 + 1;
+	while (*str1)
+	{
+		*str1++ = *str2++;
+	}
+	return (SUCCESS);
+}
+
+/**
+ * @brief Handle single quotes.
```

Comment:
Could be more detailed on what happens with single quotes. In such describing that it keeps everything inside as is and moves on to the end of the single quotes to look further for stuff to be expanded.

Reply:
True. I just wanted to push :D I'll amend

Reply:
Added better comment